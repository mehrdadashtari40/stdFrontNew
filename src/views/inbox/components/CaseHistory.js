import React from "react";
import { withTranslation } from "react-i18next";
import { Timeline, TimelineEvent } from "react-event-timeline";

function CaseHistory(props) {
  let dataTable = props.caseHistories.map((caseHistory) => (
    <TimelineEvent
      key={caseHistory.tas_uid}
      title={caseHistory.tas_title}
      icon={props.t("Task")}
      iconColor="black"
      titleStyle={{ fontWeight: 'bold' }}
    >
      {props.t("Delegated User")}:
      <p>
        <i className="fas fa-user fa-fw"> </i>
        {caseHistory.delegations[0] && caseHistory.delegations[0].usr_firstname +
            " " +
            caseHistory.delegations[0].usr_lastname
          }
      </p>
      {props.t("START DATE")}:
      <p>
        <i className="fas fa-calendar-day fa-fw"> </i>
        {caseHistory.delegations[0] &&
           caseHistory.delegations[0].del_init_date}
      </p>
      {props.t("End Date")}:
      <p>
        <i className="fas  fa-calendar-week fa-fw"> </i>
        {caseHistory.delegations[0] &&
          (caseHistory.delegations[0].del_finish_date === "Not finished"
            ? props.t("Not finished")
            : caseHistory.delegations[0].del_finish_date)}
      </p>
    </TimelineEvent>
  ));

  return (
    <Timeline
      orientation="right"
       lineColor="#fcab03"
    >
      {dataTable}
    </Timeline>
  );
}

export default withTranslation()(CaseHistory);
