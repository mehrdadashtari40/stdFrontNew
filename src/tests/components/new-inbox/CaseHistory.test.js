import React from 'react';
import { shallow } from 'enzyme';
import CaseHistory from '../../../views/inbox/components/CaseHistory';
jest.mock('react-i18next', () => ({
	withTranslation: () => Component => {
		Component.defaultProps = { ...Component.defaultProps, t: () => '' };
		return Component;
	},
}));
test('Make Snapshot file , CaseHistory Component', () => {
	let fakeCaseHistory = [
		{
			tas_uid: '3232592545e575a11045055091238715',
			tas_title: 'Task 2',
			tas_description: '',
			tas_start: 0,
			tas_type: 'NORMAL',
			tas_derivation: 'NORMAL',
			tas_assign_type: 'SELF_SERVICE',
			usr_uid: '00000000000000000000000000000001',
			usr_username: 'admin',
			usr_firstname: 'Administrator',
			usr_lastname: 'admin',
			delegations: [
				{
					del_index: 2,
					del_init_date: '1398/12/08 14:29:38',
					del_task_due_date: '1398/12/09 14:29:30',
					del_finish_date: 'Not finished',
					del_duration: 'Not finished',
					usr_uid: '00000000000000000000000000000001',
					usr_username: 'admin',
					usr_firstname: 'Administrator',
					usr_lastname: 'admin',
				},
			],
		},
	];

	const wrapper = shallow(<CaseHistory caseHistories={fakeCaseHistory} />);
	expect(wrapper).toMatchSnapshot();
});
