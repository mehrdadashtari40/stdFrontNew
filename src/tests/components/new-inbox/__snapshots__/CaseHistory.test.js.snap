// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Make Snapshot file , CaseHistory Component 1`] = `
<div>
  <table
    className="table table-bordered"
  >
    <thead>
      <tr>
        <th />
        <th />
        <th />
        <th />
      </tr>
    </thead>
    <tbody>
      <tr
        key="3232592545e575a11045055091238715"
      >
        <td>
          Task 2
        </td>
        <td>
          Administrator admin
        </td>
        <td>
          14:29:38 0777/09/17
        </td>
        <td />
      </tr>
    </tbody>
  </table>
</div>
`;
